File: FitToFit/FitToFit.Model/Akcije.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FitToFit.Model
{
    public class Akcije
    {
        public int AkcijaId { get; set; }

        public string Naziv { get; set; } = null!;

        public DateTime DatumPocetka { get; set; }

        public DateTime DatumZavrsetka { get; set; }

        public int Iznos { get; set; }
        public string? StateMachine { get; set; }

        public virtual ICollection<AkcijeTreninzi> AkcijeTreninzis { get; set; } = new List<AkcijeTreninzi>();
    }
}

File: FitToFit/UI/fittofit_admin/lib/models/akcije.dart
import 'package:fittofit_admin/models/akcijeTreninzi.dart';
import 'package:json_annotation/json_annotation.dart';

part 'akcije.g.dart';

@JsonSerializable()
class Akcije {
  int akcijaId;
  String naziv;
  DateTime datumPocetka;
  DateTime datumZavrsetka;
  int iznos;
  String? stateMachine;
  List<AkcijeTreninzi>? items = [];

  Akcije(
      {required this.akcijaId,
      required this.naziv,
      required this.datumPocetka,
      required this.datumZavrsetka,
      required this.iznos,
      this.stateMachine,
      this.items});

  factory Akcije.fromJson(Map<String, dynamic> json) => _$AkcijeFromJson(json);

  Map<String, dynamic> toJson() => _$AkcijeToJson(this);
}
